arch/arm/src/lpc17xx_40xx/lpc17_40_wdt.c

```diff
@@ -0,0 +1,358 @@
+ *           "lower-half" driver state structure.
+ *
+ * Returned Value:
+ *   Zero on success; a negated errno value on failure.
+ *
+ ****************************************************************************/
+
+static int lpc17_40_keepalive(FAR struct watchdog_lowerhalf_s *lower)
+{
+  FAR struct lpc17_40_lowerhalf_s *priv =
+      (FAR struct lpc17_40_lowerhalf_s *)lower;
+  irqstate_t flags;
+
+  /* Reload the WDT. */
+
+  flags = enter_critical_section();
+  putreg32(WDT_FEED_KEY_1, LPC17_40_WDT_FEED);
+  putreg32(WDT_FEED_KEY_2, LPC17_40_WDT_FEED);
+  priv->lastreset = clock_systime_ticks();
+  leave_critical_section(flags);
+
+  return OK;
+}
+
+/****************************************************************************
+ * Name: lpc17_40_getstatus
+ *
+ * Description:
+ *   Get the current watchdog timer status.
+ *
+ * Input Parameters:
+ *   lower  - A pointer to the publicly visible representation of the
+ *            "lower-half" driver state structure.
+ *   status - The location to return the watchdog status information.
+ *
+ * Returned Value:
+ *   Zero on success; a negated errno value on failure.
+ *
+ ****************************************************************************/
+
+static int lpc17_40_getstatus(FAR struct watchdog_lowerhalf_s *lower,
+                              FAR struct watchdog_status_s *status)
+{
+  FAR struct lpc17_40_lowerhalf_s *priv =
+      (FAR struct lpc17_40_lowerhalf_s *)lower;
+  uint32_t ticks;
+  uint32_t elapsed;
+
+  DEBUGASSERT(priv);
+
+  /* Return the status bit. */
+
+  status->flags = WDFLAGS_RESET;
+  if (priv->started)
+    {
+      status->flags |= WDFLAGS_ACTIVE;
+    }
+
+  /* Return the actual timeout in milliseconds. */
+
+  status->timeout = priv->timeout;
+
+  /* Get the elapsed time since the last ping. */
+
+  ticks   = clock_systime_ticks() - priv->lastreset;
+  elapsed = (int32_t)TICK2MSEC(ticks);
+
+  if (elapsed > priv->timeout)
+    {
+      elapsed = priv->timeout;
+    }
+
+  /* Return the approximate time until the watchdog timer expiration. */
+
+  status->timeleft = priv->timeout - elapsed;
+
+  return OK;
+}
+
+/****************************************************************************
+ * Name: lpc17_40_settimeout
+ *
+ * Description:
+ *   Set a new timeout value.
+ *
+ * Input Parameters:
+ *   lower   - A pointer to the publicly visible representation of the
+ *             "lower-half" driver state structure.
+ *   timeout - The new timeout value in milliseconds.
+ *
+ * Returned Value:
+ *   Zero on success; a negated errno value on failure.
+ *
+ ****************************************************************************/
+
+static int lpc17_40_settimeout(FAR struct watchdog_lowerhalf_s *lower,
+                               uint32_t timeout)
+{
+  FAR struct lpc17_40_lowerhalf_s *priv =
+      (FAR struct lpc17_40_lowerhalf_s *)lower;
+  uint64_t wdt_clk;
+  uint64_t wdtc;
+
+  priv->timeout = timeout;
+
+#ifdef LPC176x
```

Comment:
This lower case "x" needs to be changed to upper case: LPC176X, probably it was there before the nxstyle enforcement

Reply:
I think that this does not belong to this PR. The definition lives elsewhere (it was not introduced in this PR) and I think it will affect lots of other drivers too.  

Also, I guess all other definitions of this family would need fixing too...

Reply:
Yes, it why I said: "probably it was there"